### ISSUE TITLE
Add rule: no-any-in-generic

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#5]
## Problem

LLMs use `Array<any>`, `Record<string, any>`, `Map<string, any>` as escape hatches.

## Rule

Disallow `any` as a generic type argument.

### ❌ Bad
```ts
const items: Array<any> = [];
const cache: Record<string, any> = {};
```

### ✅ Good
```ts
const items: Array<unknown> = [];
const cache: Record<string, User> = {};
```

## Why it matters

- Forces explicit typing
- Prevents silent `any` leakage
- Encourages proper type definitions
